boot/setefibootvars_nosb.go

```diff
@@ -19,8 +20,20 @@
 
 package boot
 
-import "github.com/snapcore/snapd/osutil"
+import (
+	"errors"
+)
 
-func SetEfiBootVariables(description string, assetPath string, optionalData []byte) error {
-	return osutil.ErrDarwin
+func setEfiBootVariablesImpl(description string, assetPath string, optionalData []byte) error {
+	return errors.New("unimplemented")
+}
+
+var SetEfiBootVariables = setEfiBootVariablesImpl
+
+func MockSetEfiBootVariables(f func(description string, assetPath string, optionalData []byte) error) func() {
+	old := SetEfiBootVariables
+	SetEfiBootVariables = f
+	return func() {
+		SetEfiBootVariables = old
+	}
```

Comment:
I suppose this file could even become boot/setefibootvars.go without the `nosecboot` tag, with:
```
var SetEfiBootVariables = func(description string, assetPath string, optionalData []byte) error {
	return errors.New("unimplemented")
}

func MockSetEfiBootVariables(f func(description string, assetPath string, optionalData []byte) error) func() {
	...
}
```

and the secboot variant could have:
```
// -*- Mode: Go; indent-tabs-mode: t -*-
//go:build !nosecboot

...
func init() {
	SetEfiBootVariables = setEfiBootVariablesImpl
}

func setEfiBootVariablesImpl(description string, assetPath string, optionalData []byte) error {
	...
}
```

should avoid the chances of types going out of sync